server/s3_copy_object_action.cc

```diff
@@ -221,6 +221,43 @@ void S3CopyObjectAction::fetch_source_object_info_failed() {
 // Validate source bucket and object
 void S3CopyObjectAction::validate_copyobject_request() {
   s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
+  auto meta_directive_hdr =
+      request->get_header_value("x-amz-metadata-directive");
+  auto tagging_directive_hdr =
+      request->get_header_value("x-amz-tagging-directive");
+
+  if (!meta_directive_hdr.empty()) {
+    s3_log(S3_LOG_DEBUG, stripped_request_id,
+           "Received x-amz-metadata-directive header, value: %s",
+           meta_directive_hdr.c_str());
+    if (meta_directive_hdr == DirectiveValueReplace) {
```

Comment:
HTTP standard states case insensitivity

Reply:
I tried to use "copy" with AWS endpoint for CopyObject API, it fails with "InvalidArgument":

# aws --**endpoint-url https://s3.amazonaws.com** --profile aws s3api copy-object --bucket amitbk --copy-source archive-drone-pune-west/file-with-tags --key copy3 **--metadata-directive copy**

An error occurred (**InvalidArgument**) when calling the CopyObject operation: Unknown metadata directive.


# aws --endpoint-url https://s3.amazonaws.com --profile aws s3api copy-object --bucket amitbk --copy-source archive-drone-pune-west/file-with-tags --key copy3 **--metadata-directive COPY**
{
    "CopyObjectResult": {
        "LastModified": "2020-12-29T06:45:07.000Z",
        "ETag": "\"0f343b0931126a20f133d67c2b018a3b\""
    }
}